src/streamlink/plugins/mbcmini.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+import logging
+import re
+import json
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import useragents
+
+log = logging.getLogger(__name__)
+
+
+class Mbcmini(Plugin):
+
+    _url_re = re.compile(r'https?://miniplay\.imbc\.com/WebLiveURL\.ashx\?channel=(?P<channel_id>[a-z]+)')
+
+    _api_url = '''http://miniplay.imbc.com/WebLiveURL.ashx?channel={0}&protocol=M3U8\
+&agent=android&androidversion=18&callback=jarvis.miniInfo.loadOnAirComplete'''
+
+    @classmethod
+    def can_handle_url(cls, url):
+        return cls._url_re.match(url) is not None
+
+    def _get_streams(self):
+        match = self._url_re.match(self.url)
+        channel_id = match.group("channel_id")
+        json_url_key = "AACLiveURL"
+
+        headers = {
+            'User-Agent': useragents.CHROME,
+            'Referer': '''http://mini.imbc.com/webapp_v3/minifloat.html?\
+src=http://www.imbc.com/broad/radio/minimbc/index.html&ref='''
+        }
+        url = self._api_url.format(channel_id)
+
+        res = self.session.http.get(url, headers=headers)
+
+        json_text = '{' + ''.join(res.text.split()[1:-1])
+        json_data = json.loads(json_text)
+
+        if json_url_key in json_data:
+            return self.session.streams(json_data[json_url_key])
```

Comment:
You are recursively calling the plugin's `_get_streams` method by calling `self.session.streams` ([1](https://github.com/streamlink/streamlink/blob/6de76fca8f66dac8a9c686a5fabc6b65de1fe43e/src/streamlink/session.py#L449-L458), [2](https://github.com/streamlink/streamlink/blob/6de76fca8f66dac8a9c686a5fabc6b65de1fe43e/src/streamlink/plugin/plugin.py#L280-L317)).

In `Plugin._get_streams`, you are supposed to return an iterator or a list of objects which implement the [`Stream`](https://github.com/streamlink/streamlink/blob/6de76fca8f66dac8a9c686a5fabc6b65de1fe43e/src/streamlink/stream/stream.py) class, eg. `HLSStream`, `DashStream`, `HTTPStream`, `RTMPStream`, etc., depending on what kind of streams the website you're writing the plugin for provides.

Reply:
Hi, thanks for the review. I modified it to use HTTPStream.